cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -281,15 +296,16 @@ func newMachineController(
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeclient, 0)
 	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicclient, 0, metav1.NamespaceAll, nil)
 
-	// TODO(alberto): let environment variable to override defaultMachineAPI
-	machineDeploymentResource, _ := schema.ParseResourceArg(fmt.Sprintf("machinedeployments.%v", defaultMachineAPI))
+	specifiedMachineAPI := getAPIVersionGroup()
+
+	machineDeploymentResource, _ := schema.ParseResourceArg(fmt.Sprintf("machinedeployments.%v", specifiedMachineAPI))
 
-	machineSetResource, _ := schema.ParseResourceArg(fmt.Sprintf("machinesets.%v", defaultMachineAPI))
+	machineSetResource, _ := schema.ParseResourceArg(fmt.Sprintf("machinesets.%v", specifiedMachineAPI))
 	if machineSetResource == nil {
 		panic("MachineSetResource")
```

Comment:
This should really return an error.

Reply:
mm why? atm `MachineDeployment`, `MachineSets and `Machine` are intentionally assumed to exist so it seems reasonable to me to panic here?

Reply:
Because it makes unit tests easier to write and assert for expected behaviour.

Reply:
ok make sense, let's address this separately from this PR. I'll move the capi resource expectations into its own unit tested `func`